torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -333,53 +344,65 @@ void RequestCallbackImpl::processRpc(
             responseFuture->setError(e.what());
           }
         });
-      }
+      });
+
       return;
     }
     case MessageType::PYTHON_RREF_FETCH_CALL: {
       auto& prf = static_cast<PythonRRefFetchCall&>(rpc);
       auto& ctx = RRefContext::getInstance();
-      c10::intrusive_ptr<OwnerRRef> rref = ctx.getOwnerRRef(prf.rrefId());
-      if (rref->hasValue()) { // optional fast-path
-        auto value = rref->getValue();
-        py::object pyValue;
-        {
-          pybind11::gil_scoped_acquire ag;
-          pyValue = torch::jit::toPyObject(std::move(value));
+
+      auto futureOwner = ctx.getOwnerRRef(prf.rrefId());
+
+      if (futureOwner->completed()) { // optional fast-path
+        // the OwnerRRef has been created
+        auto& rref = futureOwner->wait();
+        if (rref->hasValue()) {
+          auto value = rref->getValue();
+          py::object pyValue;
```

Comment:
Nit: Should we move this python conversion/serialization to someplace like rpc/utils.cpp?

Reply:
Sure, will do that in a followup PR.

Reply:
added in #37249